### PR TITLE
Fix status Enforced status when Accepted condition is False

### PR DESCRIPTION
**Pull request by Boomatang**
closes: https://github.com/Kuadrant/kuadrant-operator/issues/588

# Verification Steps
- Create cluster `make local-setup`
- Create kuadrant CR
```sh
echo "
apiVersion: kuadrant.io/v1beta1
kind: Kuadrant
metadata:
  name: kuadrant
spec: {} 
" | kubectl apply -f -
```
- Create a gateway
``` sh
echo " 
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gw-1
  namespace: istio-system
  annotations:
    kuadrant.io/namespace: kuadrant-system
    networking.istio.io/service-type: ClusterIP
spec:
  gatewayClassName: istio
  listeners:
    - name: websites
      port: 80
      protocol: HTTP
      hostname: '*.website'
      allowedRoutes:
        namespaces:
          from: All
    - name: apis
      port: 80
      protocol: HTTP
      hostname: '*.io'
      allowedRoutes:
        namespaces:
          from: All
" | kubectl apply -f -
```
- Create a Authpolicy. The `Enforced` status will be false as it is not fully setup, and not need to be.
``` sh
echo "
apiVersion: kuadrant.io/v1beta2
kind: AuthPolicy
metadata:
  name: gw-1
  namespace: istio-system
spec:
  targetRef:
    group: gateway.networking.k8s.io
    kind: Gateway
    name: gw-1
  rules:
    authentication:
      "api-key-users":
        apiKey:
          selector:
            matchLabels:
              app: toystore
          allNamespaces: true
        credentials:
          authorizationHeader:
            prefix: APIKEY
    response:
      success:
        dynamicMetadata:
          "identity":
            json:
              properties:
                "userid":
                  selector: auth.identity.metadata.annotations.secret\.kuadrant\.io/user-id
" | kubectl apply -f -
```
- Set up watch on the authPolicy
```sh
watch kubectal get AuthPolicy gw-1 -n istio-system -o=jsonpath='{.status}'
```
- Delete the gateway
``` sh
kubectl delete Gateway gw-1 -n istio-system
```
- Expected result: In the watch the status block for the `Enforced` should be removed.

### CODE REVIEW

File: controllers/dnspolicy_status_test.go [hunk 1]
```diff
@@ -149,3 +154,59 @@ func TestPropagateRecordConditions(t *testing.T) {
...
+					Status: v1alpha1.DNSPolicyStatus{
+						Conditions: []metav1.Condition{
+							{
+								Message: "not accepted",
+								Type:    string(gatewayapiv1alpha2.PolicyConditionAccepted),
+								Status:  metav1.ConditionFalse,
+								Reason:  string(gatewayapiv1alpha2.PolicyReasonTargetNotFound),
+							},
+							{
+								Message: "AuthPolicy has been successfully enforced",
```
Comment (guicassolato) [inline controllers/dnspolicy_status_test.go:182]: nit:

```suggestion
								Message: "DNSPolicy has been successfully enforced",
```

File: controllers/ratelimitpolicy_status_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+					Status: kuadrantv1beta2.RateLimitPolicyStatus{
+						Conditions: []metav1.Condition{
+							{
+								Message: "not accepted",
+								Type:    string(gatewayapiv1alpha2.PolicyConditionAccepted),
+								Status:  metav1.ConditionFalse,
+								Reason:  string(gatewayapiv1alpha2.PolicyReasonTargetNotFound),
+							},
+							{
+								Message: "AuthPolicy has been successfully enforced",
```
Comment (guicassolato) [inline controllers/ratelimitpolicy_status_test.go:40]:
```suggestion
								Message: "RateLimitPolicy has been successfully enforced",
```

File: controllers/dnspolicy_status_test.go (hunk 1 above)
Comment (Boomatang) [reply controllers/dnspolicy_status_test.go:182]: Copy and paste for the win, I will change it.

File: controllers/authpolicy_status_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+							{
+								Message: "AuthPolicy has been successfully enforced",
+								Type:    string(kuadrant.PolicyConditionEnforced),
+								Status:  metav1.ConditionTrue,
+								Reason:  string(kuadrant.PolicyConditionEnforced),
+							},
+						},
+					},
+				},
+				specErr: kuadrant.NewErrInvalid("placeholder", errors.New("placeholder")),
```
[the remark this answers is not in the document]
Comment (Boomatang) [reply controllers/authpolicy_status_test.go:50]: I can change this again, The first time the CI was giving out to me saying the words I used were woke.

**Review by guicassolato — APPROVED** [review]

**Pull request merged**